src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/component/SNSPublisher.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import software.amazon.awssdk.services.sns.SnsClient
+import software.amazon.awssdk.services.sns.model.PublishRequest
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.EventDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.SNSService
+
+@Component
+class SNSPublisher(
+  private val client: SnsClient,
+  private val objectMapper: ObjectMapper,
+  @Value("\${aws.sns.enabled}") private val enabled: Boolean,
+  @Value("\${aws.sns.topic.arn}") private val arn: String,
+) {
+  fun publish(event: EventDTO) {
+    if (enabled) {
+      return buildRequestAndPublish(event)
```

Comment:
Should we log that it wasn't enabled just to help in debugging going forward?
Don't think return is necessary here

Reply:
i'd be happy with a log event at DEBUG level if the publisher is disabled. any higher level i think would be too much noise.